packages/driver/src/util/serialization/log.ts

```diff
@@ -415,7 +415,10 @@ export const preprocessLogForSerialization = (logAttrs) => {
 export const reifyLogFromSerialization = (logAttrs) => {
   let { snapshots, ... logAttrsRest } = logAttrs
 
-  if (snapshots) {
+  // if the protocol is enabled, we don't need to reify the snapshot since the snapshot was serializable coming into the primary instance of Cypress.
+  // also make sure numTestsKeptInMemory is 0, otherwise we will want to preprocess the snapshot
+  // (the driver test's set numTestsKeptInMemory to 1 in run mode to verify the snapshots)
+  if (snapshots && !(Cypress.config('protocolEnabled') && Cypress.config('numTestsKeptInMemory') === 0)) {
```

Comment:
Would be nice to have a unit test harness to add a test for this - setting up an e2e for it would be monumental!